primitives/src/json.rs

```diff
@@ -0,0 +1,201 @@
+//! String containing serialized valid JSON.
+//! This string is guaranteed to parse as JSON
+
+#[cfg(not(feature = "std"))]
+use alloc::{
+    borrow::ToOwned,
+    string::{String, ToString},
+    vec::Vec,
+};
+use core::{
+    fmt::{Display, Formatter},
+    str::FromStr,
+};
+#[cfg(feature = "std")]
+use std::{
+    string::{String, ToString},
+    vec::Vec,
+};
+
+use iroha_schema::IntoSchema;
+use parity_scale_codec::{Decode, Encode};
+use serde::{de::DeserializeOwned, Deserialize, Serialize};
+use serde_json::Value;
+
+#[allow(missing_docs)]
+#[derive(Debug, Clone, Eq, Encode, Decode)]
+pub struct JsonString(pub String);
+
+/// A candidate for a valid `JsonString`.
+/// Is used for generalizing ser/de any types to `JsonString` and vise versa
+#[derive(Serialize, Deserialize, Clone)]
+struct JsonCandidate<T>(T);
+
+impl JsonString {
+    /// Construct [`JsonString`]
+    pub fn new<T: Serialize>(payload: T) -> Self {
+        JsonCandidate(payload).try_into().unwrap()
+    }
+
+    /// Tries cast [`JsonString`] to any value.
+    ///
+    /// # Errors
+    /// - if String == invalid Json type
+    pub fn try_into_any<T: DeserializeOwned>(&self) -> Result<T, serde_json::Error> {
```

Comment:
I get that you can't make a blanket impl but I don't like API like this. Let's consider if we have some other options?

Reply:
When we discussed this with @Stukalov-A-M there we not much other options.

1. `impl<T: DeserializeOwned> TryFrom<JsonString> for T` is not possible due to `T` non-local type
2. `impl<T: DeserializeOwned> TryInto<T> for JsonString` has conflicts
3. we might introduce additional trait like `MyTryFrom` but this would require to import this trait (not sure about benefits)
4. introduce helper `Wrapper<T>` and implement `impl<T: DeserializeOwned> TryFrom<JsonString> for Wrapper<T>` (again seems not that usable)
5. we might have direct method (this case)

Reply:
I was thinking it should be th other way around. Every type that wants to be parseable from `JsonString` should implement `TryFrom<JsonString>`. It should be responsibility of the type not of `JsonString`. But that may be inconvenient?

Reply:
It would be a lot of repetition + we can't access private private fields of the `JsonString` to implement `TryFrom`